### PR TITLE
Adding pypy notifications and optimizing workflow and parsers

### PR DESCRIPTION
**Pull request by shilovmaksim**
# Description
Current pull request contains following changes:
- Added notifications about PyPy new versions not added to image
- Reworked workflow:
  - Optimized with strategy matrix
  - Considered to use message image from corresponding tool GitHub project avatar
- Optimizing verify-new-tool-version-added-to-image.ps1:
  - Sorting parsers moved out of the script as a separate modules
  - Python and PyPy parsers combined
- Improving output for workflow failure notification
  - Notification message updated with failed jobs
- Removing unused code from the previous notifications
#### Run example: 
https://github.com/shilovmaksim/versions-package-tools/actions/runs/1393929493

#### Message example: 
image

## Check list
- [x] Related issue / work item is attached
- [ ] Tests are written (if applicable)
- [ ] Documentation is updated (if applicable)

### CODE REVIEW

File: .github/workflows/get-tools-new-versions.yml [hunk 1]
```diff
@@ -9,74 +9,55 @@ defaults:
...
+          echo "::set-output name=failed-jobs::$failedJobs"
       - uses: ./.github/actions/send-slack-notification
+        name: Send Slack notification about failure
         with:
           url: ${{ secrets.SLACK_CHANNEL_URL }}
-          tool-name: 'Python or Xamarin'
+          tool-name: 'Tool name'
           pipeline-url: '$env:GITHUB_SERVER_URL/$env:GITHUB_REPOSITORY/actions/runs/$env:GITHUB_RUN_ID'
-          text: "The build of the Xamarin or Python detection pipeline has failed stages:"
+          text: "Detection pipeline for the new versions of a tools that not added to the toolset has failed jobs:/n/t${{ steps.get-failed-jobs.outputs.failed-jobs }}"
```
Comment (miketimofeev) [inline .github/workflows/get-tools-new-versions.yml:63]:
```suggestion
          text: "Missing toolset tool versions checker pipeline has failed jobs:/n/t${{ steps.get-failed-jobs.outputs.failed-jobs }}"
```

File: get-new-tool-versions/parsers/verify-added-to-image/verify-python-parser.psm1 [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+        [string]$ReleasesUrl,
+        [string]$FilterParameter,
+        [string]$FilterArch
+    )
+    
+    $stableReleases = (Invoke-RestMethod $ReleasesUrl) |
+        Where-Object stable -eq $true |
+        ForEach-Object {$_.$FilterParameter.split(".")[0,1] -join"."} |
+        Select-Object -Unique
+    $toolsetUrl = "https://raw.githubusercontent.com/actions/virtual-environments/main/images/win/toolsets/toolset-2019.json"
```
Comment (miketimofeev) [inline get-new-tool-versions/parsers/verify-added-to-image/verify-python-parser.psm1:13]: let's check windows-2022 as it's the latest one
```suggestion
    $toolsetUrl = "https://raw.githubusercontent.com/actions/virtual-environments/main/images/win/toolsets/toolset-2022.json"
```

File: get-new-tool-versions/parsers/verify-added-to-image/verify-python-parser.psm1 [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    $latestExistingMinorVesion = ((Invoke-RestMethod $toolsetUrl).toolcache |
+        Where-Object {$_.name -eq $ToolName -and $_.arch -eq $FilterArch}).versions |
+        ForEach-Object {$_.split(".")[0,1] -join"."} |
+        Select-Object -Last 1
```
Comment (miketimofeev) [inline get-new-tool-versions/parsers/verify-added-to-image/verify-python-parser.psm1:17]: suggest rewriting it to use two variables for readability and take the latest one from the start
```suggestion
    $latestVersion = ((Invoke-RestMethod $toolsetUrl).toolcache | Where-Object {$_.name -eq $ToolName -and $_.arch -eq $FilterArch}).versions | Select-Object -Last 1
    $latestMinorVesion = $LatestVersion.Split(".")[1]
```

Comment (shilovmaksim) [reply get-new-tool-versions/parsers/verify-added-to-image/verify-python-parser.psm1]: Agree. Updated

File: .github/workflows/get-tools-new-versions.yml (hunk 1 above)
Comment (shilovmaksim) [reply .github/workflows/get-tools-new-versions.yml]: Updated

File: get-new-tool-versions/parsers/verify-added-to-image/verify-python-parser.psm1 (hunk 2 above)
Comment (shilovmaksim) [reply get-new-tool-versions/parsers/verify-added-to-image/verify-python-parser.psm1]: Done

File: get-new-tool-versions/parsers/verify-added-to-image/verify-xamarin-parser.psm1 [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+    $xamarinReleases = (Invoke-RestMethod $ReleasesUrl).items
+    $filteredReleases = $xamarinReleases | Where-Object {$_.name -in $FilterProducts.name} | Sort-Object name | Select-Object name, version
+    $toolsetUrl = "https://raw.githubusercontent.com/actions/virtual-environments/main/images/macos/toolsets/toolset-11.json"
+    $uploadedReleases = (Invoke-RestMethod $toolsetUrl).xamarin
+    $releasesOnImage = @()
+    foreach ($FilterProduct in $FilterProducts) {
+        $releasesOnImage += @{$FilterProduct.name = $uploadedReleases.($FilterProduct.property)}
+    }
+    $versionsToAdd = $filteredReleases | Where-Object {$releasesOnImage.($_.name) -notcontains $_.version } | ForEach-Object {[string]::Empty} {
```
Comment (miketimofeev) [inline get-new-tool-versions/parsers/verify-added-to-image/verify-xamarin-parser.psm1:15]: nit:
```suggestion
    $versionsToAdd = $filteredReleases | Where-Object {$releasesOnImage.($_.name) -notcontains $_.version} | ForEach-Object {[string]::Empty} {
```

Comment (shilovmaksim) [reply get-new-tool-versions/parsers/verify-added-to-image/verify-xamarin-parser.psm1:15]: Sigh

**Review by MaksimZhukov — APPROVED** [review]

**Review by miketimofeev — APPROVED** [review]

**Pull request merged**